### ISSUE TITLE
identify the div number of the place where <span type="omissionEnd"/> occurs

### ISSUE THREAD

**Issue by arlogriffiths** [erc-dharma/tfd-nusantara-philology#65]
Dear @ajaniak —

1. BUG. The sections svayambhu_06.19.02 through svayambhu_06.22.02 are affected by omissions, of varying extent, in two of the three witnesses. To my surprise, the `<span type="omissionEnd"/>` of the omission in witness `#K` is used to define the end of the omission in `#L` in the display below, and none of the apparatus entries occurring between `<span type="omissionStart"/>` and `<span type="omissionEnd"/>` in `#L` show the autogenerated `<rdg>` "om. L" that is expected. Since the first `<span type="omissionStart"/>` occurs in `#K` and we do get autogenerated "om. K" where we expect them, it seems that the code cannot handle the co-occurrence of two gaps in the same part of the text.

2. ENHANCEMENT. With the present display, we don't help the reader see in which section the `<span type="omissionEnd"/> ` occurs. 

<img width="1152" alt="Capture d’écran 2022-05-06 à 10 16 13" src="https://user-images.githubusercontent.com/835603/167094285-381d1331-e154-4c0e-9273-ce3f64c209a9.png">

Could we change the autogenerated lemma in the display to become something like this (after the above bug has been solved)?

**kalih ... (§86) ikaṅ**

**Comment by ajaniak** [comment]
The enhancement part is done. (and it should be already applied) 

The bug is partially resolved : the fake lemma are generated correctly, but I still need to work on the Javascript to resolve the incorrect display regarding om and lac (larger gap).

**Comment by ajaniak** [comment]
I have updated the code to resolve the 1. The code might take several hours to be udpated and the pipeline will need to be then relaunched. 

Here results from the tests
<img width="1104" alt="Capture d’écran 2022-07-12 à 10 09 37" src="https://user-images.githubusercontent.com/33328906/178442141-56f8316a-ee0f-4845-a073-5dd8f095b3b0.png">
<img width="1329" alt="Capture d’écran 2022-07-12 à 10 04 19" src="https://user-images.githubusercontent.com/33328906/178442153-88900faf-a509-4ca6-a077-1118513f4842.png">

**Comment by ajaniak** [comment]
To follow on our conversion regarding the context when creating  `§xx` with `<ptr>` as well as for the lemma of larger gap. 

I have look into it today, if the possibility to declare in the `<teiHeader>` using `<refsDecl>` is possible. It seems a complex solution and I don't think it will be easy to set up for most encoders without a XML-TEI expert. So I wonder for the longevity of the guide, if it is a good idea. 

For the `<ptr>`, the context can be deduced from the content of `@target` since the encoder can provide the necessary level. I have updated the display this morning to match `§xx`, but we could fine a solution between the previous and the new to clear state the context: `§name xx` name standing for a label we could create based on `@type` and elements, i.e `lg` becoming `stanza` and xx being the number.  

For the larger gap situation, I have added a few supplementary rules regarding the display of `(§xx)` 
- if the beginning and the end of the omission are in the same bloc container, we don't display the (§xx) in the lemma. Case in Candrakirana
- if the beginning and the end of the omission are not in the same bloc container,
     - If the context is the dyad, only its number is given. Situation in svayambhu
     - If the context if a paragraph inside a div, the number of the main div and the paragraph are given, i.e. §11.3. Case in Candrakirana and Saṅ Hyaṅ Siksa Kandaṅ Karəsian

Since, the display offers satisfaying results for those files, I wonder if we could avoid using the solution we evoked with `<refsDecl>`. 

For now, I have worked mostly for omission, and I will need to update the reformulation and lacunae at some point. If you agree with the first result and if it matches your expectations.
